client-react/src/dashboard/CategoryBox.js

```diff
@@ -20,17 +24,57 @@ const Title = styled.h1`
   font-size: 25pt;
 `;
 
+const ContentWrapper = styled.div`
+  margin-left: 30px !important;
+  margin: 10px;
+  display: ${(props) => (props.display ? "block" : "none")};
+`;
+
+const IconWrapper = styled.span`
+  display: ${(props) => (props.display ? "block" : "none")};
+`;
+
 const CategoryBox = ({ title }) => {
+  const [display, setDisplay] = useState([true, false]);
+
+  const toggleCategory = (index) => {
+    setDisplay(
+      display.map((x, i) => {
+        if (index !== i) {
+          x = true;
+        } else {
+          x = false;
+        }
+        return x;
+      })
+    );
+  };
+
   return (
     <CategoryWrapper>
       <TitleWrapper>
         <div style={{ display: "flex", alignItems: "center" }}>
-          <FontAwesomeIcon icon={faCaretRight} size="lg" />
+          <IconWrapper display={display[0] ? true : false}>
+            <FontAwesomeIcon
+              icon={faCaretRight}
+              size="lg"
+              onClick={() => toggleCategory(0)}
+            />
+          </IconWrapper>
+          <IconWrapper display={display[1] ? true : false}>
+            <FontAwesomeIcon
+              icon={faCaretDown}
+              size="lg"
+              onClick={() => toggleCategory(1)}
+            />
+          </IconWrapper>
           <Title>{title}</Title>
         </div>
         <FontAwesomeIcon icon={faEllipsisH} color="#707070" />
       </TitleWrapper>
-      <div>PLACE HOLDER FOR REST OF CONTENT</div>
+      <ContentWrapper display={display[1] ? true : false}>
+        <div style={{ height: "50px" }}>PLACE HOLDER FOR REST OF CONTENT</div>
```

Comment:
Is there a specific reason why each component needs to check an index in the state to determine if it should render? Based on what was written, I think conditional rendering might simplify the logic.

Reply:
An onclick doesn't refresh the conditional rendering right? I just tried and it isn't working as intended. If you have a way you want me to implement it, maybe discussing it would be better